dialogue-annotations-processor/src/main/java/com/palantir/dialogue/annotations/processor/data/HttpPathParser.java

```diff
@@ -17,14 +17,16 @@
 package com.palantir.dialogue.annotations.processor.data;
 
 import com.google.common.base.Splitter;
-import java.util.ArrayList;
 import java.util.List;
 import java.util.Optional;
+import java.util.stream.Collectors;
 import javax.lang.model.element.Element;
-import org.glassfish.jersey.uri.internal.UriTemplateParser;
 
 public final class HttpPathParser {
 
+    private static final Splitter SPLITTER =
+            Splitter.on('/').omitEmptyStrings(); // omit empty segments; typically the first segment is empty
```

Comment:
This may change behavior when a pair of slashes is expected -- can we normalize out a leading slash instead?

Reply:
Is this not equivalent to the existing `segment.isEmpty()` check?

Happy to change this, since I agree that only omitted empty segments on the first segment is better behavior. Just want to make sure I correctly understand what the code is currently doing.

Reply:
I'm a little worried about changing the behavior here if an existing path template has repeated slashes.

Reply:
oh hmm, I'd missed that. I agree that we shouldn't change behavior here, I can revisit this later.